### PR TITLE
jenkins-pipeline update

### PR DESCRIPTION
**Pull request by szpak**
Also dependency refreshment.

### CODE REVIEW

File: src/test/resources/jobs/boot-microservice-run-smoke-tests-for-current-version.xml [hunk 1]
```diff
@@ -38,13 +38,16 @@
...
       <fromRootBuildScriptDir>true</fromRootBuildScriptDir>
+      <useWorkspaceAsHome>false</useWorkspaceAsHome>
     </hudson.plugins.gradle.Gradle>
   </builders>
   <publishers>
     <hudson.tasks.junit.JUnitResultArchiver>
       <testResults>**/build/test-results/**/*.xml</testResults>
       <keepLongStdio>false</keepLongStdio>
       <testDataPublishers/>
+      <allowEmptyResults>false</allowEmptyResults>
```
Comment (tkrzeminski) [inline src/test/resources/jobs/boot-microservice-run-smoke-tests-for-current-version.xml:49]: Probably this should be set to `true` as `JUnit` not always generates data.

File: src/test/resources/jobs/boot-microservice-run-smoke-tests-for-previous-production-version.xml [hunk 2]
```diff
@@ -61,33 +58,38 @@ return [GIT_CHECKOUT_REF: gitRef]</groovyScriptContent>
...
         <org.jenkins__ci.plugins.flexible__publish.ConditionalPublisher>
-          <condition class="org.jenkins_ci.plugins.run_condition.contributed.ShellCondition">
-            <command>[ ! -z "${PREV_APP_VERSION}" ]</command>
-          </condition>
           <publisherList>
             <hudson.tasks.junit.JUnitResultArchiver>
               <testResults>**/build/test-results/**/*.xml</testResults>
               <keepLongStdio>false</keepLongStdio>
               <testDataPublishers/>
+              <allowEmptyResults>false</allowEmptyResults>
```
Comment (tkrzeminski) [inline src/test/resources/jobs/boot-microservice-run-smoke-tests-for-previous-production-version.xml:78]: Should be `true`

**Review by tkrzeminski — APPROVED** [review]

File: src/test/resources/jobs/boot-microservice-run-smoke-tests-for-current-version.xml (hunk 1 above)
Comment (s2lomon) [reply src/test/resources/jobs/boot-microservice-run-smoke-tests-for-current-version.xml:49]: I think that's the default value here.

Comment (szpak) [reply src/test/resources/jobs/boot-microservice-run-smoke-tests-for-current-version.xml:49]: In Groovy code there is:
```
                        conditionalbuilders {
                            'hudson.tasks.junit.JUnitResultArchiver' {
                                testResults(DEFAULT_GRADLE_JUNIT_XML_REPORTS_PATH)
                                keepLongStdio('false')
                                testDataPublishers()
                            }
                        }
```

Comment (szpak) [reply src/test/resources/jobs/boot-microservice-run-smoke-tests-for-current-version.xml:49]: We may change it if desired (in general yes).

**Review by s2lomon — APPROVED** [review]
Looks ok for me

**Pull request merged**